src/gui/inputdevices/InputUtils.cpp

```diff
@@ -22,13 +22,15 @@ bool InputUtils::applyButton(ToolHandler* toolHandler, Settings* settings, Butto
     return toolChanged;
 }
 
-void InputUtils::warnIfQuestionableTouchDrawingSettings(ToolHandler* toolHandler, Settings* settings) {
+bool InputUtils::touchDrawingDisallowed(ToolHandler* toolHandler, Settings* settings) {
     ButtonConfig* cfg = settings->getButtonConfig(Button::BUTTON_TOUCH);
     if (cfg->getDisableDrawing() && cfg->getAction() == TOOL_NONE && toolHandler->isDrawingTool()) {
         g_message("Ignoring touchscreen for drawing.\n"
```

Comment:
Could you modify this to only ever print this message once per instance? Otherwise the console might be flooded and printing to the console is quite slow in comparison.

Reply:
Done :)